pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaManaged.hpp

```diff
@@ -359,33 +359,131 @@ class StateVectorCudaManaged
-                    "`ctrls` and `ctrls_values` must have the same size.");
+                    "`ctrls` and `controlled_values` must have the same size.");
         auto ctrlsInt = NormalizeCastIndices<std::size_t, int>(
             controlled_wires, BaseType::getNumQubits());
         auto tgtsInt = NormalizeCastIndices<std::size_t, int>(
             tgt_wires, BaseType::getNumQubits());
         auto ctrls_valuesInt =
             Pennylane::Util::cast_vector<bool, int>(controlled_values);
 
-        if (opName == "GlobalPhase") {
-            const std::vector<std::string> names(tgt_wires.size(), "I");
+        if (opName == "MultiRZ") {
+            const std::vector<std::string> names(tgtsInt.size(), {"RZ"});
+            applyParametricPauliGeneralGate_(names, ctrlsInt, ctrls_valuesInt,
+                                             tgtsInt, params.front(), adjoint);
+        } else if (opName == "GlobalPhase") {
+            const std::vector<std::string> names(tgtsInt.size(), "I");
             applyParametricPauliGeneralGate_(names, ctrlsInt, ctrls_valuesInt,
                                              tgtsInt, 2 * params[0], adjoint);
+
+        } else if (native_gates_.find(opName) != native_gates_.end()) {
+            applyParametricPauliGeneralGate_({opName}, ctrlsInt,
+                                             ctrls_valuesInt, tgtsInt,
+                                             params.front(), adjoint);
+        } else if (opName == "Rot") {
+            if (adjoint) {
+                auto rot_matrix =
+                    cuGates::getRot<CFP_t>(params[2], params[1], params[0]);
+                applyDeviceGeneralGate_(rot_matrix.data(), ctrlsInt, tgtsInt,
+                                        ctrls_valuesInt, true);
+            } else {
+                auto rot_matrix =
+                    cuGates::getRot<CFP_t>(params[0], params[1], params[2]);
+                applyDeviceGeneralGate_(rot_matrix.data(), ctrlsInt, tgtsInt,
+                                        ctrls_valuesInt, false);
+            }
+        } else if (par_gates_.find(opName) != par_gates_.end()) {
+            // TODO: offload to par_gates_ if available
+            auto &gateMap =
+                cuGates::DynamicGateDataAccess<PrecisionT>::getInstance();
+            auto &&matrix_cu = gateMap.getGateData(opName, params);
+
+            gate_cache_.add_gate(opName, params[0], matrix_cu);
+
+            applyDeviceGeneralGate_(
+                gate_cache_.get_gate_device_ptr(opName, params[0]), ctrlsInt,
+                tgtsInt, ctrls_valuesInt, adjoint);
+        } else { // No offloadable function call; defer to matrix passing
+            auto &&par =
+                (params.empty()) ? std::vector<Precision>{0.0} : params;
+            if (!gate_cache_.gateExists(opName, par[0]) &&
+                gate_matrix.empty()) {
+                std::string message = "Currently unsupported gate: " + opName;
```

Comment:
"Currently unsupported gate: " + opName + " and no matrix is provided"?
```suggestion
                std::string message = "Currently unsupported gate: " + opName + " and no matrix is provided";
```